docs: rework README

lucab opened the pull request:
This moves docpages around and re-arranges the README so that it
provides a direct high-level overview of the project, putting all
the finer details inside sub-pages.

---

[Rendered](https://github.com/lucab/afterburn/blob/401fb2a58e4c60afb260a9072eb6fd88c22069c7/README.md)

miabbott commented on docs/contributing.md:
```diff
@@ -16,6 +16,8 @@ contribution. See the [DCO](DCO) file for details.
 
 Due to their public nature, GitHub and mailing lists are not appropriate places for reporting vulnerabilities. Please refer to CoreOS's [security disclosure][disclosure] process when reporting issues that may be security related.
 
+[disclosure]: https://coreos.com/security/disclosure/
```
Should this be updated to point to Red Hat disclosure docs?

bgilbert replied:
Probably.  The CoreOS disclosure process is going away.

lucab replied:
Ah, good catch. I think I can drop this section at all, the repository already has the "security notice page" set and that is properly pointing toward Red Hat.

bgilbert commented on README.md:
```diff
@@ -4,140 +4,64 @@
-    - Attributes
-      - AFTERBURN_VAGRANT_VIRTUALBOX_PRIVATE_IPV4
-      - AFTERBURN_VAGRANT_VIRTUALBOX_HOSTNAME
+Depending on the specific platform, the following services may run in the [initramfs](./dracut/30afterburn/) on first-boot:
```
`first boot`

bgilbert commented on README.md:
```diff
@@ -4,140 +4,64 @@
+The following features are conditionally available on some platforms as [systemd service units](./systemd/):
+ * installing public SSH keys for local system users
+ * retrieving [attributes](./docs/usage/attributes.md) from instance metadata
+ * checking-in to the provider in order to report a successful boot or instance provisioning 
```
`checking in`

bgilbert reviewed (approved):
LGTM!

Commit:
docs: rework README

This moves docpages around and re-arranges the README so that it
provides a direct high-level overview of the project, putting all
the finer details inside sub-pages.
